test/parallel/test-https-client-get-url.js

```diff
@@ -31,7 +29,12 @@ var server = https.createServer(options, function(req, res) {
 });
 
 server.listen(common.PORT, function() {
-  https.get('https://127.0.0.1:' + common.PORT + '/foo?bar');
+  https.get({
+    host: '127.0.0.1',
+    port: common.PORT,
+    path: '/foo?bar',
+    rejectUnauthorized: false
+  });
 });
```

Comment:
Yeah, I see what you mean. OTOH having the undocumented environment variable that globally disables certificate checks for this one test seems janky.

Here's the code in `https.js`:
```js  if (typeof options === 'string') {
    options = url.parse(options);
    if (!options.hostname) {
      throw new Error('Unable to determine the domain name');
    }
  } else {
    options = util._extend({}, options);
  }
```

Maybe we can leave it as `https.get(<object>)` here and test URL parsing separately? We have some tests in `test/parallel/test-http-url.parse-*.js`

I can also add a case that calls 'https.get()` on a path with no hostname, and check that it throws the right exception.